internal/storage/bucket_handle.go

```diff
@@ -63,17 +63,9 @@ func (bh *bucketHandle) NewReader(
 		obj = obj.Generation(req.Generation)
 	}
 
-	// Creating a NewRangeReader instance.
-	r, err := obj.NewRangeReader(ctx, start, length)
-	if err != nil {
-		err = fmt.Errorf("error in creating a NewRangeReader instance: %v", err)
-		return
-	}
-
-	// Converting io.Reader to io.ReadCloser by adding a no-op closer method
-	// to match the return type interface.
-	rc = io.NopCloser(r)
-	return
+	// Returning NewRangeReader instance.
```

Comment:
Instead we can club both comments in one - 
"NewRangeReader creates a "storage.Reader" object which is also a io.ReadCloser since it contains both Read() and Close() methods present in io.ReadCloser interface."

Reply:
fixed